src/component/dataZoom/AxisProxy.js

```diff
@@ -245,9 +241,31 @@ AxisProxy.prototype = {
             percentWindow[idx] = boundPercent;
         });
 
+        asc(valueWindow);
+        asc(percentWindow);
+
+        // The windows from user calling of `dispatchAction` might be out of the extent,
+        // or do not obey the `min/maxSpan`, `min/maxValueSpan`. But we dont restrict window
+        // by `zoomLock` here, because we see `zoomLock` just as a interaction constraint,
+        // where API is able to initialize/modify the window size even though `zoomLock`
+        // specified.
+        var spans = this._minMaxSpan;
+        hasPropModeValue
+            ? restrictSet(valueWindow, percentWindow, dataExtent, percentExtent, false)
+            : restrictSet(percentWindow, valueWindow, percentExtent, dataExtent, true);
+
+        function restrictSet(fromWindow, toWindow, fromExtent, toExtent, toValue) {
+            var suffix = toValue ? 'Span' : 'ValueSpan';
+            sliderMove(0, fromWindow, fromExtent, 'all', spans['min' + suffix], spans['max' + suffix]);
+            for (var i = 0; i < 2; i++) {
+                toWindow[i] = numberUtil.linearMap(fromWindow[i], fromExtent, toExtent, true);
+                toValue && (toWindow[i] = scale.parse(toWindow[i]));
```

Comment:
I suppose this line should be  `toValue && (toWindow[i] = scale.parse(toValue[i]))` ?